src/client/Microsoft.Identity.Client/Internal/Requests/ClientCredentialRequest.cs

```diff
@@ -31,7 +31,7 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             MsalAccessTokenCacheItem cachedAccessTokenItem = null;
             var logger = AuthenticationRequestParameters.RequestContext.Logger;
 
-            if (!_clientParameters.ForceRefresh)
+            if (!_clientParameters.ForceRefresh && !AuthenticationRequestParameters.HasClaims)
```

Comment:
Should HasClaims be changed to HaveClaims, so it reads as AuthenticationRequestParameters.HaveClaims? Or if it refers to the request, for consistency, then it's fine. (Just a thought, I'm eh about it.)

Reply:
In my mind, it's the request that has claims. Naming is so hard :)